gdnative-core/src/core_types/color.rs

```diff
@@ -47,6 +48,68 @@ impl Color {
             a: self.a + (weight * (other.a - self.a)),
         }
     }
+    #[inline]
+    pub fn blend(&self, other: &Color) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_blend)(self.sys(), other.sys()) })
+    }
+
+    #[inline]
+    pub fn contrasted(&self) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_contrasted)(self.sys()) })
+    }
+    #[inline]
+    pub fn darkened(&self, amount: f32) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_darkened)(self.sys(), amount) })
+    }
+    #[inline]
+    pub fn from_hsv(h: f32, s: f32, v: f32, a: f32) -> Color {
+        let color = Color::rgba(0.0, 0.0, 0.0, 0.0);
+        Color::from_sys(unsafe { (get_api().godot_color_from_hsv)(color.sys(), h, s, v, a) })
+    }
+    #[inline]
+    pub fn gray(&self) -> f32 {
+        // Implemented as described in godot docs
+        (self.r + self.b + self.g) / 3.0
+    }
+
+    #[inline]
+    pub fn inverted(&self) -> Color {
+        // Implementation as described in godot docs.
+        Color {
+            r: 1.0f32 - self.r,
+            g: 1.0f32 - self.g,
+            b: 1.0f32 - self.b,
+            a: self.a,
+        }
+    }
+
+    pub fn to_abgr32(&self) -> i32 {
```

Comment:
`u32` is a bit more common for byte representations, it's `i32` mostly because GDScript doesn't have unsigned types. Since people are likely going to do shift arithmetic, `u32` is probably more intuitive...

Reply:
Thanks for the feedback. The reason that I initially used i32 is because that is what the internal `to_abgr32()` function returns. I'll go ahead and add the conversion to u32

Reply:
> I'd be careful with casting i32 to [u8; 4] due to the endianness problem.

This should be OK when using shift operations and not bit reinterpretation.
The question is more: which type is more useful?

Reply:
> which type is more useful?

Hmm, honestly I'm not sure. I think `[u8; 4]` is better if you wanted to process the color, but why do that when you had full floating point precision to begin with? Perhaps some weird blending operators like XOR, but I never really needed that, and if you wanted to blend a lot of colors you should really be using a shader. If you just want to put it into a file or something, `u32` would suffice.

Reply:
Ok, let's use `u32` then.

Maybe it's even better to leave byte conversion to the user: he can then use standard `u32::to_ne_bytes()` for reinterpretation (native byte order), or one of the Little/Big Endian variants.

Reply:
On the other hand, when using an array it's clear what the bytes are and the whole endianess topic doesn't arise in the first place. Serialization might be a bit more work, but also safer due to that. 

And the argument I brought above can also be made the other way around, with `u32::from_ne_bytes()` etc.

Reply:
I don't have a strong opinion either way. I think `[u8; 4]` is also fine when implemented with the standard conversion methods.

Reply:
I think what might be nice is that `[u8; 4]` makes two things immediately clear:
* Each component is in 0..255 range, and integral (not float)
* The order of array elements follows the method name (whereas an integer would need to be taken apart, and this can be done wrong)

Similar for `[u16; 4]`: it's immediately clear how the data is represented.

Reply:
OK, I think `[u8; 4]` is fine, If it turns out to be a bad decision, we can always change that later on.